src/status_im/signing/core.cljs

```diff
@@ -203,6 +204,8 @@
                                   :formatted-data (if typed? (types/json->clj data) (ethereum/hex->text data))
                                   :keycard-step (when pinless? :connect)})}
        (when pinless?
+         (keycard.card/start-nfc {:on-success #(re-frame/dispatch [:keycard.callback/start-nfc-success])
```

Comment:
It might never happen, but if `keycard.card/start-nfc` will be blocked somewhere outside js env and `signing.keycard/hash-message` will be called before `keycard.card/start-nfc`. Can this be an issue? If so we should move `signing.keycard/hash-message` to the callback. Also it makes sense if there can be failure.

Reply:
@rasom it shouldn't happen, but even if it happens, that hash-message call is calling status-go, so in that case it would hash the message without sending it to the card.

Reply:
> that hash-message call is calling status-go
yes but it's callback calls keycard, right? what happen if we call keycard even if `keycard.card/start-nfc` fails?

Reply:
@rasom it calls keycard but just asking to sign when nfc is ready and when the user taps the card, so in that case it won't do anything

Reply:
understood